### ISSUE TITLE
Bogus group update messages sent each time when opening the group

### ISSUE THREAD

**Issue by sivaraam** [signalapp/Signal-Android#10344]
- [x] I have searched open and closed issues for duplicates
- [x] I am submitting a bug report for existing functionality that does not work as intended
- [x] I have read https://github.com/signalapp/Signal-Android/wiki/Submitting-useful-bug-reports
- [x] This isn't a feature request or a discussion topic

----------------------------------------

### Bug description
I'm in the Signal [beta testing group](https://community.signalusers.org/t/beta-testing-the-new-gv2-groups-big-signal-group-with-as-many-volunteering-forum-members-as-possible/17030). I recently switched phones and restored from the backup in my new device. For some time now, each time I open the group a bogus group updated message is sent to the members even though I haven't updated anything. Also, these updates aren't visible in my conversation history. This is happening consistently for me but I don't see such messages from other members of the group.

Note: Arguably, one could stalk my visits to the group using these bogus update messages 😉 

### Steps to reproduce
Incidentally, this doesn't seem to be happening for everyone but for me the steps are:
1. Open Signal
2. Open the beta testing group

**Actual result:** Bogus group updated message is sent to other members of the group
**Expected result:** No such bogus messages are sent to the group members

### Screenshots
![signal-2020-12-25-121248](https://user-images.githubusercontent.com/12448084/103235694-ba8a8a00-4968-11eb-8e47-a35215065861.jpg)

### Device info

**Device:** Nokia 2.3
**Android version:** 10
**Signal version:** 5.0.8

### Link to debug log
https://debuglogs.org/d84a006d87b5ff3b5fb03024d3d41da2c95dd66b25f8ce111d478e7aa8a5c451 (taken shortly after the opening the group)

**Comment by u32i64** [comment]
I'm also a member of said `__...group_v2...56`. Noticed similar log lines repeating with successive group revisions in my log which I think are likely related:

```
I GroupsV2StateProcessor: Applying P2P group change
I GroupsV2StateProcessor: Saved server query for group change
W GroupStateMapper: Could not find group log on server V670
I GroupsV2StateProcessor: Skipping profile sharing detection as was already a full member before update
I GroupsV2StateProcessor: Skipping profile sharing detection as was already a full member before update
[...]
I RecipientDatabase: Persisting 76 Profile keys, 1 of which are authoritative
[...]
I RecipientDatabase: Profile key from owner RecipientId::23
```

My full debug log: https://debuglogs.org/08daa45ee2c6a6da574fb1c609813be40ee0fe7531696660ddd168739b20d9b8

In @sivaraam's log, I've noticed that Signal is indeed trying to update profile key of self:

```
I JobRunner: [JOB::bb4bf1ac-d560-48f7-afe8-af9b39b7fa9a][GroupV2UpdateSelfProfileKeyJob][1] Running job. (Time Since Submission: 8075 ms, Lifespan: 86400000 ms, Run Attempt: 2/Unlimited)
I GroupV2UpdateSelfProfil: Updating profile key on group __...group_v2...56
[...]
I Job: [JOB::9219d48d-8b34-4a7c-a436-d84b8a50414f][PushGroupSilentUpdateSendJob] onSubmit() (Time Since Submission: 18 ms, Lifespan: 86400000 ms, Run Attempt: 1/Unlimited)
I JobRunner: [JOB::bb4bf1ac-d560-48f7-afe8-af9b39b7fa9a][GroupV2UpdateSelfProfileKeyJob][1] Job finished with result SUCCESS in 1232 ms. (Time Since Submission: 9307 ms, Lifespan: 86400000 ms, Run Attempt: 2/Unlimited)
I JobRunner: [JOB::9219d48d-8b34-4a7c-a436-d84b8a50414f][PushGroupSilentUpdateSendJob][3] Running job. (Time Since Submission: 31 ms, Lifespan: 86400000 ms, Run Attempt: 1/Unlimited)
```

**Comment by Disk1of5** [comment]
I am having this issue with with a family member running non-beta, and all that was done was re-storing backup to a new phone. this happens for all group conversations even newly created ones.

**Comment by alan-signal** [comment]
@Disk1of5 new phone with new number or same number? If same number, did they use their PIN successfully?

**Comment by alan-signal** [comment]
@sivaraam Can you open the group again and get me another log please?

**Comment by alan-signal** [comment]
Please confirm it is fixed in 5.1 if you're in the beta program.

**Comment by sivaraam** [comment]
> Please confirm it is fixed in 5.1 if you're in the beta program.

Yeah. I can confirm that this looks fixed in 5.1. Thanks!

**Comment by bentolor** [comment]
@alan-signal I'm currently stumbling over the same situation and I think i debugged the issue. It's due to the following constellation:

1. A v2 group with member Alice, Bob and Carl
2. Alice has disabled read notifications and especially envelopes from anybody
3. Carl has not had any direct contact with Alice.

Is this the situation fixed with https://github.com/signalapp/Signal-Android/commit/6080e1f338aa3bf3a872df4be6d526f02683c14b or am I'm guessing in the wrong direction?

These are the messages I see from Alice:

```
{"envelope":{
    "source":"+1CALLALICE", "sourceDevice":1,"relay":null,
    "timestamp":1610911648301,
    "dataMessage":{
       "timestamp":1610911648301, "message":null,  "expiresInSeconds":0,
       "reaction":null,"quote":null,"mentions":[],"attachments":[],
       "groupInfo": "groupId":"SECRETSANATA","members":null,"name":null,"type":"UPDATE"}},
       "syncMessage":null,"callMessage":null,"receiptMessage":null
}}
```
